### PR TITLE
notmuch: Add custom layout for notmuch buffers

### PR DESCRIPTION
**Pull request by Miciah**
Define a new custom layout with binding "n" for notmuch buffers.  The custom layout's name can be customized using the new `notmuch-spacemacs-layout-name` variable. The custom layout's binding can be customized using the new `notmuch-spacemacs-layout-binding` variable.

### CODE REVIEW

File: layers/+email/notmuch/config.el [hunk 1]
```diff
@@ -0,0 +1,24 @@
+;;; config.el --- notmuch Layer configuration File for Spacemacs
+;;
+;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
```
Comment (nixmaniack) [inline layers/+email/notmuch/config.el]: You should be the `Author:`.

Comment (Miciah) [reply layers/+email/notmuch/config.el]: Fixed.

**Comment by jb55** [comment]
Awesome! I was wondering why notmuch wasn't working well in a new
perspective but this seems to work much better.

**Comment by Miciah** [comment]
Rebased.

**Comment by syl20bnr** [comment]
If you can add the documentation to the `README.org`, that would be wonderful :angel:

**Comment by Miciah** [comment]
Added changes to `README.org`.

**Comment by syl20bnr** [comment]
Thank you ! 💜 

I made all the instances of custom layouts consistent in commit https://github.com/syl20bnr/spacemacs/commit/fea035f4

Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**